### ISSUE TITLE
RV32 AMOCAS.D emulated as LR/SC pair

### ISSUE THREAD

**Issue by moscickimilosz** [riscv/riscv-isa-manual#1502]
Hi!

The unprivileged specification version 20240411 states that:
> Any AMO can be emulated by an LR/SC pair

Is this with exception of AMOCAS.D for RV32? Since:
> An SC can only pair with the most recent LR in program order.

**Comment by ved-rivos** [comment]
That is correct. For RV32, `AMOCAS.D` cannot be emulated using LR/SC and `AMOCAS.Q` cannot be emulated using LR/SC. 
AMO that operate on data wider than `XLEN` cannot be emulated using an `LR`/`SC` pair.

**Comment by moscickimilosz** [comment]
Thanks for clarifying, would it make sense to note this in the spec?

**Comment by aswaterman** [comment]
I don't think a clarification is needed.
